### PR TITLE
Drop tests for Python 2.x and 3.4

### PR DESCRIPTION
**Pull request by connorworley**
These versions are EOL and the tests have been broken against them for a while. Let's just stop testing against them.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,13 @@
 language: python
 
 python:
-    - 2.7
-    - 3.4
     - 3.5
     - 3.6
-    - pypy
+    - 3.7
```
Comment (aelawson) [inline .travis.yml:6]: Should we also add 3.8?

Comment (connorworley) [reply .travis.yml:6]: I don't think it's packaged on xenial or bionic, but maybe focal. Let me see what we can do with travis.

Comment (aelawson) [reply .travis.yml:6]: Cool if it's non-trivial we don't have to block this PR on it. Would be nice to have though.

Comment (connorworley) [reply .travis.yml:6]: Looks like some tests fail: https://travis-ci.org/github/dropbox/stone/jobs/697871257

**Pull request merged**